src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -96,12 +107,39 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
 
         await this._paypalCommercePaymentProcessor.initialize(paramsScript, undefined, gatewayId);
 
+        const fundingKey = this._paypalCommerceFundingKeyResolver.resolve(methodId, gatewayId);
+
+        const buttonContainer = document.querySelector(container);
+        if (!buttonContainer) {
+            throw new PaymentMethodClientUnavailableError("Unable to initialize Paypal Commerce because container element wasn't found in the DOM");
+        }
+
         this._paypalCommercePaymentProcessor.renderButtons(cartId, container, buttonParams, {
             onRenderButton,
-            fundingKey: this._paypalCommerceFundingKeyResolver.resolve(methodId, gatewayId),
+            fundingKey,
             paramsForProvider: {isCheckout: true},
         });
 
+        if (this._isAPM)  {
+            const apmFieldsContainerId = apmFieldsContainer || `${container}apmFieldsContainer`;
+            const fullName = `${firstName} ${lastName}`;
+
+            if (!document.querySelector(apmFieldsContainerId)) {
+                const apmFieldsContainerElement = document.createElement('div');
```

Comment:
Per our discussion, it is worthy to remove behavior that implicitly creates a container in the case when it is not given in options. The APM feature should be fully controllable by the options of strategy and in the case if consumers with custom checkout update `checkout-SDK, they will need to follow our docs in order to make APM feature work in their custom checkout implementation